logger.js

```diff
@@ -54,19 +62,30 @@ function pinoLogger (opts, stream) {
       return
     }
 
-    log[level]({
-      res: this,
-      responseTime: responseTime
-    }, 'request completed')
+    if (this.shouldLogSuccess) {
+      log[level]({
+        res: this,
+        responseTime: responseTime
+      }, 'request completed')
+    }
   }
 
   function loggingMiddleware (req, res, next) {
     req.id = genReqId(req)
     req.log = res.log = logger.child({req: req})
     res[startTime] = res[startTime] || Date.now()
-    if (!req.res) { req.res = res }
 
     if (autoLogging) {
+      // calculate here whether to log success, but apply this flag up to the last minute.
+      // this is to ensure that the event listener still executes (as there may be other logic that is needed).
+      var shouldLogSuccess = true
+      if (req.url && autoLoggingIgnorePaths.length) {
+        for (let i = 0; shouldLogSuccess && i < autoLoggingIgnorePaths.length; i++) {
+          shouldLogSuccess = req.url.match(autoLoggingIgnorePaths[i]) == null
+        }
+      }
+
+      res.shouldLogSuccess = shouldLogSuccess
       res.on('finish', onResFinished)
```

Comment:
Instead of keeping some state around, why can't you just avoid to add this listener?

Reply:
Ok, i can do that. I was explaining that I didn't want to cut off future logic in that listener, but thats ok. let me push again.

Reply:
I think it would have less impact on the overhead if we avoid to add that listener.

Reply:
fixed

Reply:
however, do note that the listener will still execute the this.removeListener('finished') even when that listener was not registered. It doesn't fail, however it doesn't seem quite right.

So I divided onResError and onResFinished, but refactoring further did not work as expected. When I tried refactoring into `getResponseTime` and `getLogLevel`, and passing the `this` (response), one test failed ('err.raw is available to custom serializers').

Reply:
I think this approach is correct, don't mind for the double removeListener, it's very fast anyway.